hermeto/core/errors.py

```diff
@@ -104,6 +104,20 @@ class NotAGitRepo(PackageRejected):
     """A package turned out to be not a git repository."""
 
 
+class PackageWithCorruptLockfileRejected(PackageRejected):
+    """Package lock file does not match package config."""
+
+    def __init__(self, package_path: str) -> None:
+        """Initialize the error."""
+        reason = f"{package_path} contains a lock file that does not match corresponding Cargo.toml"
```

Comment:
nitpick: this is a backend specific error the whole purpose of which is to wrap the parent with a better msg string. Even the string itself, by the nature of containing a reference to Cargo.toml, suggests that it would likely be better placed in the cargo backend package.